src/platform/w3dengine/client/w3ddebugicons.cpp

```diff
@@ -0,0 +1,263 @@
+                        int frame_delta = icon->endFrame - frame;
+
+                        if (frame_delta < 1) {
+                            any_ended = true;
+                        } else {
+
+                            if (frame_delta < fade_start) {
+                                alpha = (int)((float)frame_delta / (float)fade_start * (float)alpha);
+                            }
+
+                            RGBColor color = icon->color;
+
+                            float offset = icon->width / 2.0f;
+
+                            int diffuse = (alpha << 24) | color.Get_As_Int();
+
+                            float x = icon->position.x;
+                            float y = icon->position.y;
+                            float z = icon->position.z;
+
+                            verts->x = x - offset;
+                            verts->y = y - offset;
+                            verts->z = z;
+                            verts->diffuse = diffuse;
+                            verts->u1 = 0.0f;
+                            verts->v1 = 0.0f;
+                            ++verts;
+
+                            verts->x = x + offset;
+                            verts->y = y - offset;
+                            verts->z = z;
+                            verts->diffuse = diffuse;
+                            verts->u1 = 0.0f;
+                            verts->v1 = 0.0f;
+                            ++verts;
+
+                            verts->x = x + offset;
+                            verts->y = y + offset;
+                            verts->z = z;
+                            verts->diffuse = diffuse;
+                            verts->u1 = 0.0f;
+                            verts->v1 = 0.0f;
+                            ++verts;
+
+                            verts->x = x - offset;
+                            verts->y = y + offset;
+                            verts->z = z;
+                            verts->diffuse = diffuse;
+                            verts->u1 = 0.0f;
+                            verts->v1 = 0.0f;
+                            ++verts;
+
+                            *indices++ = unsigned short(vertex_count + 0);
+                            *indices++ = unsigned short(vertex_count + 1);
+                            *indices++ = unsigned short(vertex_count + 2);
+
+                            *indices++ = unsigned short(vertex_count + 0);
+                            *indices++ = unsigned short(vertex_count + 2);
+                            *indices++ = unsigned short(vertex_count + 3);
+
+                            index_count += 6;
+
+                            vertex_count += 4;
+                        }
+                        ++index;
+                    }
+                }
+
+                if (vertex_count == 0) {
+                    break;
+                }
+
+                DX8Wrapper::Set_Shader(ShaderClass(0x984B7));
```

Comment:
What does this magic number represent

Reply:
a shader but yeah should have a todo to resolve